lib/util.js

```diff
@@ -0,0 +1,170 @@
+'use strict'
+
+const { URL } = require('url')
+const Ref = require('json-schema-resolver')
+
+function addHook (fastify) {
+  const routes = []
+  const sharedSchemasMap = new Map()
+
+  fastify.addHook('onRoute', (routeOptions) => {
+    routes.push(routeOptions)
+  })
+
+  fastify.addHook('onRegister', async (instance) => {
+    // we need to wait the ready event to get all the .getSchemas()
+    // otherwise it will be empty
+    instance.addHook('onReady', (done) => {
+      const allSchemas = instance.getSchemas()
+      for (const schemaId of Object.keys(allSchemas)) {
+        if (!sharedSchemasMap.has(schemaId)) {
+          sharedSchemasMap.set(schemaId, allSchemas[schemaId])
+        }
+      }
+      done()
+    })
+  })
+
+  return {
+    routes,
+    Ref () {
+      const externalSchemas = Array.from(sharedSchemasMap.values())
+      return Ref({ clone: true, applicationUri: 'todo.com', externalSchemas })
```

Comment:
Why the hardcoded `applicationUri`?

Reply:
This is done to resolve relative $refs when the user (wrongly) doesn't set it.
Can't remember why I set it like this.. need to give it a check!

Reply:
I will just left a comment to address the issue. It is not really related to this PR.